ibis/expr/types.py

```diff
@@ -192,14 +194,24 @@ def execute(self, limit='default', params=None, **kwargs):
           Pass an integer to effect a specific row limit. limit=None means "no
           limit". The default is whatever is in ibis.options.
 
+        timecontext: Tuple(pd.Timestamp, pd.Timestamp) or None, default None
```

Comment:
The property type should be ` Tuple[pd.Timestamp, pd.Timestamp]`

Reply:
I add a typing in typing.py that define timecontext as
```SupportsTimestamp = Union[pd.Timestamp, str]```
```TimeContext = Optional[Tuple[SupportsTimestamp, SupportsTimestamp]]```

Reply:
We should just do `Optional[Tuple[pd.Timestamp, pd.Timestamp]]`, internally this should never be str

Reply:
Actually, maybe we just disallow str all together. If we want to allow passing begin, end as str we can add it later.

For simplicity, I think we probably should just only support pd.Timestamp for now

Reply:
OK, I will make it simple then, just updated it to allow pd.Timetstamp only.